src/game_events/conditional_wml.cpp

```diff
@@ -212,16 +212,38 @@ bool conditional_passed(const vconfig& cond)
 	return matches;
 }
 
-bool matches_special_filter(const config &cfg, const vconfig& filter)
+bool matches_special_filter(const_attack_ptr attack, const vconfig& filter, const map_location& a)
 {
-	if (!cfg) {
+	if (!attack) {
 		WRN_NG << "attempt to filter attack for an event with no attack data.";
 		// better to not execute the event (so the problem is more obvious)
 		return false;
 	}
+	const unit_map& units = resources::gameboard->units();
+	const unit_map::const_iterator att = units.find(a);
+	assert(att.valid());
```

Comment:
How do we know this assert won't trigger?

Given that we already found two units in the calling function, why is there an additional units.find() call in this function?

Reply:
remove that and put const unit in declaration